### PR TITLE
enhancement(dropdown): variant with an icon

### PR DESCRIPTION
**Pull request by yashasvi13**
### JIRA Ticket 

Jira:  [AN-1715](https://hello-haptik.atlassian.net/browse/AN-1715)

--

### Changelog
- Dropdown variant with a prefix icon

### Test Link
https://60c3333adf6c37000802e1b6--stoic-knuth-02312b.netlify.app/?path=/story/components-inputs-dropdown--documentation

### CODE REVIEW

File: src/components/Inputs/Dropdown/Dropdown.js [hunk 1]
```diff
@@ -381,6 +383,7 @@ const Dropdown = ({
             optionKey={optionKey}
             disabled={disabled}
             prefixClassName={`${prefixClassName}-input`}
+            prefixIcon={prefixIcon}
```
Comment (sidMulay) [inline src/components/Inputs/Dropdown/Dropdown.js:386]: Do we need `prefixIcon` for older (non-themed) dropdown?

File: src/components/Inputs/Dropdown/ThemedDropdown.js [hunk 2]
```diff
@@ -255,6 +264,14 @@ const DropdownArrow = styled("div")(
   DropdownArrowMode,
 );
 
+const DropdownPrefix = styled("div")(
```
Comment (sidMulay) [inline src/components/Inputs/Dropdown/ThemedDropdown.js:267]: - Use `Box` here and re-style it per your needs.
- Rename it to `DropdownPrefixIcon`

File: src/components/Inputs/Dropdown/components/Input/Input.js [hunk 3]
```diff
@@ -65,6 +74,16 @@ const Input = ({
...
 
+  const themedPrefixComponent = prefixIcon ? (
+    <DropdownPrefix>
+      <Icon
+        className={styles.icon}
+        prefixClassName={`${prefixClassName}-icon`}
+        src={prefixIcon}
+      />
+    </DropdownPrefix>
+  ) : null;
```
Comment (sidMulay) [inline src/components/Inputs/Dropdown/components/Input/Input.js:85]: themedPrefixIcon && <Icon .../>, no need of ternary operator here.

File: src/components/Inputs/Dropdown/components/Input/Input.js [hunk 4]
```diff
@@ -65,6 +74,16 @@ const Input = ({
     </DropdownArrow>
   );
 
+  const themedPrefixComponent = prefixIcon ? (
+    <DropdownPrefix>
+      <Icon
+        className={styles.icon}
+        prefixClassName={`${prefixClassName}-icon`}
+        src={prefixIcon}
+      />
```
Comment (sidMulay) [inline src/components/Inputs/Dropdown/components/Input/Input.js:83]: Do we need to use `className` and `prefixClassName` here?

File: src/components/Inputs/Dropdown/components/Input/Input.scss [hunk 5]
```diff
@@ -11,4 +11,8 @@
       border-top-right-radius: 0;
     }
   }
+  .icon {
+    margin-right: 4px;
+  }
```
Comment (sidMulay) [inline src/components/Inputs/Dropdown/components/Input/Input.scss:16]: We don't need to make changes in scss file. Make the style changes in in `DropdownPrefix` via Design System. Refer to `DropdownArrow`

**Review by sidMulay — CHANGES_REQUESTED** [review]

**Review by sidMulay — DISMISSED** [review]

**Review by gopal-virtual — APPROVED** [review]

**Review by sidMulay — APPROVED** [review]

**Pull request merged**